lib/utils/Utils.ts

```diff
@@ -0,0 +1,343 @@
+  }
+}
+
+/**
+ * Turns the ntoken transform data into a valid json array
+ * @param data
+ */
+export function refineNTokenData(data: string) {
+  return data
+    .replace(/function\(d,e\)/g, '"function(d,e)').replace(/function\(d\)/g, '"function(d)')
+    .replace(/function\(\)/g, '"function()').replace(/function\(d,e,f\)/g, '"function(d,e,f)')
+    .replace(/\[function\(d,e,f\)/g, '["function(d,e,f)').replace(/,b,/g, ',"b",')
+    .replace(/,b/g, ',"b"').replace(/b,/g, '"b",').replace(/b]/g, '"b"]')
+    .replace(/\[b/g, '["b"').replace(/}]/g, '"]').replace(/},/g, '}",')
+    .replace(/""/g, '').replace(/length]\)}"/g, 'length])}');
+}
+
+/**
+ *
+ */
+export function uuidv4() {
+  if (getRuntime() === 'node') {
+    return Reflect.get(module, 'require')('crypto').webcrypto.randomUUID();
+  }
+
+  if (globalThis.crypto?.randomUUID()) {
+    return globalThis.crypto.randomUUID();
+  }
+
+  // See https://stackoverflow.com/a/2117523
+  return '10000000-1000-4000-8000-100000000000'.replace(/[018]/g, (cc) => {
+    const c = parseInt(cc);
+    return (c ^ window.crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16);
+  });
+
+
+}
+
+export type Runtime = 'node' | 'deno' | 'browser';
+
+/**
+ *
+ */
+export function getRuntime(): Runtime {
+  if ((typeof process !== 'undefined') && (process?.versions?.node))
+    return 'node';
+  if (Reflect.has(globalThis, 'Deno'))
+    return 'deno';
+  return 'browser';
+}
+
+/**
+ *
+ */
+export function isServer() {
+  return [ 'node', 'deno' ].includes(getRuntime());
+}
+
+/**
+ *
+ * @param stream
+ */
+export async function* streamToIterable(stream: ReadableStream<Uint8Array>) {
+  const reader = stream.getReader();
+  try {
+    while (true) {
+      const { done, value } = await reader.read();
+      if (done) {
+        return;
+      }
+      yield value;
+    }
+  } finally {
+    reader.releaseLock();
+  }
+}
+
+export const debugFetch: FetchFunction = (input, init) => {
+  const url =
+        typeof input === 'string' ?
+          new URL(input) :
+          input instanceof URL ?
+            input :
+            new URL(input.url);
+
+
+  const headers =
+        init?.headers ?
+          new Headers(init.headers) :
+          input instanceof Request ?
+            input.headers :
+            new Headers();
+
+  const arr_headers = [ ...headers ];
+
+  console.log(
+    'YouTube.js Fetch:\n' +
+        `  url: ${url.toString()}\n` +
+        `  method: ${init?.method || 'GET'}\n` +
+        `  headers:\n${
+          arr_headers.length > 0 ?
+            `${arr_headers.map(([ key, value ]) => `    ${key}: ${value}`).join('\n')}\n` :
+            '    (none)\n'
+        }  body:\n${
+          init?.body ? `${typeof init.body === 'string' ?
+            headers.get('content-type') === 'application/json' ?
+              JSON.stringify(JSON.parse(init.body), null, 2) :
+              init.body :
+            '    <binary>'
+          }` : '    (none)'}`
+  );
```

Comment:
This is fine but it might be better to move the logic out of this and have the print statement be easy to read.